huaweicloud/config/auth.go

```diff
@@ -339,6 +347,42 @@ func buildClientByPassword(c *Config) error {
 	return genClients(c, projectAuthOptions, domainAuthOptions)
 }
 
+func buildClientByAgency(c *Config) error {
+	client, err := NewIamClient(c, "")
+	if err != nil {
+		return fmt.Errorf("Error creating Huaweicloud IAM client: %s", err)
+	}
+
+	request := &iam_model.CreateTemporaryAccessKeyByAgencyRequest{}
+	domainNameAssumeRoleIdentityAssumerole := c.AssumeRoleDomain
+	durationSecondsAssumeRoleIdentityAssumerole := int32(24 * 60 * 60)
```

Comment:
suggest to define it as a const variable.

Reply:
done